components/brave_rewards/core/ledger_impl.h

```diff
@@ -7,16 +7,28 @@
 #define BRAVE_COMPONENTS_BRAVE_REWARDS_CORE_LEDGER_IMPL_H_
 
 #include <map>
-#include <memory>
 #include <queue>
 #include <string>
 #include <utility>
 
 #include "base/containers/flat_map.h"
 #include "base/types/always_false.h"
 #include "brave/components/brave_rewards/common/mojom/bat_ledger.mojom.h"
+#include "brave/components/brave_rewards/core/api/api.h"
```

Comment:
And please remove
```
namespace braveledger_media {
class Media;
}
```
too. Thanks!

Reply:
Removed the class forward declaration.

About the inclusion files, which ones are you suggesting can be removed? `api.h` is required, as we have `api::API api_` as a class member. I have checked all the other files and they seemed to be used in the header itself.

Reply:
Thanks @cdesouza-chromium!
I meant the ones you introduced in `ledger_impl.h`, e.g.
```
#include "brave/components/brave_rewards/core/api/api.h"
```
is now present in both `ledger_impl.h` and `ledger_impl.cc`.

Reply:
Done.

Just as a heads up though, the style guide has been updated on this regard, suggesting the opposite, to avoid IWYU violations.

https://google.github.io/styleguide/cppguide.html#Include_What_You_Use

Reply:
Interesting. Well, I guess we should follow upstream conventions then. Don't change your PR just because of that, though (unless you have to for some other reason). Making a note to myself to add them back.